api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java

```diff
@@ -77,6 +77,14 @@ public String name() {
     return name;
   }
 
+  public TableIdentifier toLowerCase() {
+    String[] newLevels = Arrays.stream(namespace().levels())
+        .map(String::toLowerCase)
```

Comment:
nit: Would it be a good idea to pass an explicit `Locale.ROOT` to `toLowerCase`?

Reply:
Yes, I agree.

Reply:
Actually, I think this is correct using the default locale.

We usually pass `Locale.ROOT` to avoid the [Turkish locale problem](http://mattryall.net/blog/2009/02/the-infamous-turkish-locale-bug) for strings that are known to be English, typically constants. But table names should be handled in the environment's locale because they are likely to be in the locale's language.